src/otx/algo/detection/yolov9.py

```diff
@@ -0,0 +1,196 @@
+    new_state_dict_keys: dict[str, str] = {}
+    for key in state_dict:
+        match = re.match(r"^(\d+)\.(.*)$", key)
+        if match:
+            orig_idx = int(match.group(1))
+            rest_key = match.group(2)
+
+            if orig_idx < backbone_len:
+                new_key = f"backbone.module.{orig_idx}.{rest_key}"
+            elif orig_idx < backbone_len + neck_len:
+                new_idx = orig_idx - backbone_len
+                new_key = f"neck.module.{new_idx}.{rest_key}"
+            else:  # for bbox_head
+                new_idx = orig_idx - backbone_len - neck_len
+                new_key = f"bbox_head.module.{new_idx}.{rest_key}"
+            new_state_dict_keys[key] = new_key
+
+    for old_key, new_key in new_state_dict_keys.items():
+        value = state_dict.pop(old_key)
+        state_dict[new_key] = value
+
+    super(SingleStageDetector, self)._load_from_state_dict(
+        state_dict,
+        prefix,
+        local_metadata,
+        strict,
+        missing_keys,
+        unexpected_keys,
+        error_msgs,
+    )
+
+
+class YOLOv9(ExplainableOTXDetModel):
+    """OTX Detection model class for YOLOv9.
+
+    Default input size per model:
+        - yolov9_t : (640, 640) # not supported yet
+        - yolov9_s : (640, 640)
+        - yolov9_m : (640, 640)
+        - yolov9_c : (640, 640)
+        - yolov9_e : (640, 640) # not supported yet
+    """
+
+    input_size_multiplier = 32  # TODO (sungchul): need to check
+    mean: tuple[float, float, float] = (0.0, 0.0, 0.0)
+    std: tuple[float, float, float] = (255.0, 255.0, 255.0)
+
+    def __init__(
+        self,
+        model_name: Literal["yolov9_t", "yolov9_s", "yolov9_m", "yolov9_c", "yolov9_e"],
+        label_info: LabelInfoTypes,
+        input_size: tuple[int, int] = (640, 640),
+        optimizer: OptimizerCallable = DefaultOptimizerCallable,
+        scheduler: LRSchedulerCallable | LRSchedulerListCallable = DefaultSchedulerCallable,
+        metric: MetricCallable = MeanAveragePrecisionFMeasureCallable,
+        torch_compile: bool = False,
+        tile_config: TileConfig = TileConfig(enable_tiler=False),
+    ) -> None:
+        self.load_from: str = PRETRAINED_WEIGHTS[model_name]
+        super().__init__(
+            model_name=model_name,
+            label_info=label_info,
+            input_size=input_size,
+            optimizer=optimizer,
+            scheduler=scheduler,
+            metric=metric,
+            torch_compile=torch_compile,
+            tile_config=tile_config,
+        )
+
+    def _build_model(self, num_classes: int) -> SingleStageDetector:
+        backbone = GELAN(model_name=self.model_name)
+        neck = YOLONeck(model_name=self.model_name)
+        bbox_head = YOLOHead(model_name=self.model_name, num_classes=num_classes)
+
+        # patch _load_from_state_dict
+        SingleStageDetector._load_from_state_dict = _load_from_state_dict_for_yolov9  # type: ignore[method-assign] # noqa: SLF001
```

Comment:
Instead of changing class, how about changing instance?

Reply:
If using `detector._load_from_state_dict = _load_from_state_dict_for_yolov9` instead, the instance's methods are not reachable in `_load_from_state_dict_for_yolov9` due to `self` is not given like below
https://github.com/openvinotoolkit/training_extensions/blob/e38a7aa453326274ebf25ab5fd7017e65de985e6/src/otx/algo/utils/mmengine_utils.py#L146-L154

Reply:
It can be changed as below
```
detector._load_from_state_dict = _load_from_state_dict_for_yolov9.__get__(detector, SingleStageDetector)
```
Please refer https://stackoverflow.com/questions/50599045/python-replacing-a-function-within-a-class-of-a-module

Reply:
Thanks, it works, done at 2bf9c3b